Add possibility to set dimmer step.

grzegorz914 opened the pull request.

grzegorz914 committed:
Add possibility to set dimmer step.

grzegorz914 committed:
Revert "Add possibility to set dimmer step."

This reverts commit 73908efe5130631fb995b3c24f687813e97216f1.

grzegorz914 committed:
Add possibility to set dimmer step.

grzegorz914 committed:
Add possibility to set data refresh

grzegorz914 commented:
Is this wrong place?

druciak commented:
Now the files are in the right place. But you pushed 2 features to one branch, so both are here. Would it be a problem to make two pull requests? You need to make 2 branches, each should contain one commit.
If not we can proceed here.

grzegorz914 commented:
We can proceed here.

druciak commented on src/__init__.py:
```diff
@@ -51,7 +51,8 @@ def initConfig():
     config_root.sitemap = ConfigText(default="default", fixed_size=False)
-    config_root.refresh = ConfigNumber(3)
+    config_root.refresh = ConfigSelection(default=5, choices=[1, 2, 3, 5, 10, 20])
+    config_root.dimmer = ConfigSelection(default=1, choices=[1, 2, 3, 5, 10])
```
Before default step value was 5. Please change it to 5, this way it will behave exactly as it was before.

grzegorz914 replied:
Done

druciak commented on src/plugin.py:
```diff
@@ -92,10 +92,11 @@ class SliderWidget(SitemapWidget, ConfigSlider):
     def __init__(self, item, sub_page, increment=5):
         SitemapWidget.__init__(self, item, sub_page)
-        ConfigSlider.__init__(self, default=toint(item.get("state")) if item else 0, increment=increment, limits=(SliderWidget.MIN_VAL, SliderWidget.MAX_VAL))
+        ConfigSlider.__init__(self, default=toint(item.get("state")) if item else 0, iincrement=(SliderWidget.DIMMER_STEP), limits=(SliderWidget.MIN_VAL, SliderWidget.MAX_VAL))
```
Please leave as it was and pass the configured value as "increment" parameter. Also there a typo in parameter name "iincrement".

grzegorz914 replied:
Typo fixed

druciak commented on src/plugin.py:
```diff
@@ -167,7 +168,7 @@ def __init__(self, session, sitemap):
         self.refreshTimer = eTimer()
         self.refreshTimer.callback.append(self.refresh_data)
-        self.refreshTimer.start(config_root.refresh.value*1000)
+        self.refreshTimer.start(config_root.refresh.value*100)
```
Why do you need resolution greater than seconds? What's your configured value? I don't think this is a good idea, because it pulls the whole sitemap during refresh, so it is going to kill your network. What's your motivation to this change?

grzegorz914 replied:
Ah ok, I think the plugin load map only one time on startup plugin and after this refresh only changed value, if not so I return to default value 1000ms.

druciak replied:
This would be ideal and most of OH client does update only changed values (through websockets) but this plugin is intended to be simply as much as possible. My use case is to start the plugin, turn off/on some lights in the living room and close the window. I don't thing it makes sense to build "real" OH client for such purpose.

druciak commented on src/setup.py:
```diff
@@ -33,6 +33,8 @@ def __init__(self, session):
             getConfigListEntry(_("User"), config_root.user),
             getConfigListEntry(_("Password"), config_root.password),
             getConfigListEntry(_("Sitemap"), config_root.sitemap),
+            getConfigListEntry(_("Data refresh *100ms"), config_root.refresh),
```
*100ms is confusing for the users. They don't know what "10" value means, it is not obvious that "10" means one second. It should be either seconds or milliseconds.

grzegorz914 replied:
In my branch PLC logic miliseconds is basis value, it’s not problem for me I return to 1s multiple, done.

druciak commented on src/setup.py:
```diff
@@ -33,6 +33,8 @@ def __init__(self, session):
             getConfigListEntry(_("Password"), config_root.password),
             getConfigListEntry(_("Sitemap"), config_root.sitemap),
+            getConfigListEntry(_("Data refresh *100ms"), config_root.refresh),
+            getConfigListEntry(_("Dimmer step"), config_root.dimmer),
```
Both texts should have polish translations in "po" file. Use poedit or whatever tool you like to add these entries.

grzegorz914 replied:
Done

druciak commented on src/__init__.py:
```diff
@@ -51,7 +51,8 @@ def initConfig():
     config_root.password = ConfigPassword()
     config_root.sitemap = ConfigText(default="default", fixed_size=False)
-    config_root.refresh = ConfigNumber(3)
+    config_root.refresh = ConfigSelection(default=5, choices=[1, 2, 3, 5, 10, 20])
```
With such set o values it is impossible to choose 3 seconds which the default previously. Also seem my other comments regarding refresh time.

grzegorz914 replied:
Done

druciak reviewed (changes requested):
Thank you for the contribution. See my comments and make the changes as requested. Then I will merge this PR.

grzegorz914 committed:
Update __init__.py

Updated default value for dimmer step to 5, also change refresh value to 3 as it was before.

grzegorz914 committed:
Update plugin.py

Fixed typo in increment

grzegorz914 committed:
Update setup.py

Updated config for data refresh multiple 1s

grzegorz914 committed:
Update plugin.py

Update data refresh to multiple 1s as it before

grzegorz914 committed:
Update pl.po

Updated *.po file  "Data refresh" and "Dimmer step"

grzegorz914 committed:
Update openHAB.pot

Updated *.pot file "Data refresh" and "Dimmer step"

grzegorz914 committed:
Update pl.po

Set correct source for "Dimmer step"
